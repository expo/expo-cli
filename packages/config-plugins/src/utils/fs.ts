import fs from 'fs';
import path from 'path';

/** A basic function that copies a single file to another file location. */
export async function copyFilePathToPathAsync(src: string, dest: string): Promise<void> {
  const srcFile = await fs.promises.readFile(src);
  await fs.promises.mkdir(path.dirname(dest), { recursive: true });
  await fs.promises.writeFile(dest, srcFile);
}
