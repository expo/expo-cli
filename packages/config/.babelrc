{
  "presets": [
    [
      "env",
      {
        "targets": {
          "node": "6.0.0"
        }
      }
    ]
  ],
  "plugins": [
    "transform-runtime",
    ["transform-object-rest-spread", { "useBuiltins": true }],
    "transform-flow-strip-types"
  ]
}
