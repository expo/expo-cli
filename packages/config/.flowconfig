[ignore]
<PROJECT_ROOT>/build/.*
.*/eslint-config-universe/.*

[options]
emoji=true

esproposal.decorators=ignore
esproposal.class_static_fields=enable
esproposal.class_instance_fields=enable
esproposal.export_star_as=enable

module.ignore_non_literal_requires=true

unsafe.enable_getters_and_setters=true

suppress_comment=.*\\$FlowFixMe
suppress_comment=.*\\$FlowIssue
suppress_comment=\\(.\\|\n\\)*\\$FlowNewLine
suppress_comment=\\(.\\|\n\\)*\\$ExpectError

