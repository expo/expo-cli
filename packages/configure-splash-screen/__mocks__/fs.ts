'use strict';
import { fs } from 'memfs';
module.exports = fs;
