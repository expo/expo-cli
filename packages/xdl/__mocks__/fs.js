const mockfs = require('mock-fs');

const container = {
  __internalFs: null,
  __configureFs: conf => {
    container.__internalFs = mockfs.fs(conf);
  },
  __mockFile: mockfs.file,
  __mockDirectory: mockfs.directory,
  __mockSymlink: mockfs.symlink,
};

const proxyfs = new Proxy(container, {
  get(target, property, receiver) {
    if (target.hasOwnProperty(property)) {
      return target[property];
    } else {
      if (target.__internalFs === null) {
        target.__internalFs = mockfs.fs();
      }

      return target.__internalFs[property];
    }
  },
});

module.exports = proxyfs;
