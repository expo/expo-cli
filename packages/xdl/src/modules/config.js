const defaultUniversalModuleConfig = {
  ios: {
    // subdirectory in which the module podspec is placed.
    subdirectory: 'ios',
    // whether when adding a new version of ABI
    // the module should be versioned and released
    // as eg. ABI28_0_0EXCamera
    versionable: true,
    // whether the module should be included in the newly created detached app.
    detachable: true,
    // whether the module should be included in Expo Client
    includeInExpoClient: true,
  },
  android: {
    // subdirectory in which the Android project can be found.
    subdirectory: 'android',
    // whether when adding a new version of ABI
    // the module should be versioned and released
    // as eg. abi28_2_0.expo.modules.camera
    versionable: true,
    // whether the module should be included in the newly created detached app,
    // so when releasing a new version of expoview
    // as eg. host.exp.exponent:expoview:28.2.0 the module will
    // be assembled and uploaded to the local maven repo
    detachable: true,
    // whether the module should be included in Expo Client
    includeInExpoClient: true,
  },
};

const firebaseModuleConfig = {
  ios: {
    versionable: false,
    detachable: false,
    includeInExpoClient: false,
  },
  android: {
    versionable: false,
    detachable: false,
    includeInExpoClient: false,
  },
};

const expoSdkUniversalModules = [
  // native modules
  {
    podName: 'EXAdsAdMob',
    libName: 'expo-ads-admob',
    sdkVersions: '>=30.0.0',
  },
  {
    podName: 'EXAdsFacebook',
    libName: 'expo-ads-facebook',
    sdkVersions: '>=33.0.0',
  },
  {
    podName: 'EXAmplitude',
    libName: 'expo-analytics-amplitude',
    sdkVersions: '>=33.0.0',
  },
  {
    podName: 'EXSegment',
    libName: 'expo-analytics-segment',
    sdkVersions: '>=30.0.0',
  },
  {
    podName: 'EXAppAuth',
    libName: 'expo-app-auth',
    sdkVersions: '>=32.0.0',
  },
  {
    podName: 'EXAppLoaderProvider',
    libName: 'expo-app-loader-provider',
    sdkVersions: '>=32.0.0',
  },
  {
    podName: 'EXAV',
    libName: 'expo-av',
    sdkVersions: '>=33.0.0',
  },
  {
    podName: 'EXBackgroundFetch',
    libName: 'expo-background-fetch',
    sdkVersions: '>=32.0.0',
  },
  {
    podName: 'EXBarCodeScanner',
    libName: 'expo-barcode-scanner',
    sdkVersions: '>=30.0.0',
  },
  {
    podName: 'EXBarCodeScannerInterface',
    libName: 'expo-barcode-scanner-interface',
    sdkVersions: '>=30.0.0',
  },
  {
    podName: 'EXBlur',
    libName: 'expo-blur',
    sdkVersions: '>=33.0.0',
  },
  {
    podName: 'EXBrightness',
    libName: 'expo-brightness',
    sdkVersions: '>=33.0.0',
  },
  {
    podName: 'EXCalendar',
    libName: 'expo-calendar',
    sdkVersions: '>=33.0.0',
  },
  {
    podName: 'EXCamera',
    libName: 'expo-camera',
    sdkVersions: '>=29.0.0',
  },
  {
    podName: 'EXCameraInterface',
    libName: 'expo-camera-interface',
    sdkVersions: '>=29.0.0',
  },
  {
    podName: 'EXConstants',
    libName: 'expo-constants',
    sdkVersions: '>=29.0.0',
  },
  {
    podName: 'EXConstantsInterface',
    libName: 'expo-constants-interface',
    sdkVersions: '>=29.0.0',
  },
  {
    podName: 'EXContacts',
    libName: 'expo-contacts',
    sdkVersions: '>=30.0.0',
  },
  {
    podName: 'EXCore',
    libName: 'expo-core',
    sdkVersions: '>=29.0.0',
  },
  {
    podName: 'EXDocumentPicker',
    libName: 'expo-document-picker',
    sdkVersions: '>=33.0.0',
  },
  {
    podName: 'EXErrors',
    libName: 'expo-errors',
    sdkVersions: '>=32.0.0',
  },
  {
    podName: 'EXFacebook',
    libName: 'expo-facebook',
    sdkVersions: '>=33.0.0',
  },
  {
    podName: 'EXFaceDetector',
    libName: 'expo-face-detector',
    detachable: false,
    sdkVersions: '>=29.0.0',
  },
  {
    podName: 'EXFaceDetectorInterface',
    libName: 'expo-face-detector-interface',
    sdkVersions: '>=29.0.0',
  },
  {
    podName: 'EXFacebook',
    libName: 'expo-facebook',
    sdkVersions: '>=33.0.0',
  },
  {
    podName: 'EXFileSystem',
    libName: 'expo-file-system',
    sdkVersions: '>=29.0.0',
  },
  {
    podName: 'EXFileSystemInterface',
    libName: 'expo-file-system-interface',
    sdkVersions: '>=29.0.0',
  },
  {
    podName: 'EXFirebaseAnalytics',
    libName: 'expo-firebase-analytics',
    sdkVersions: '>=31.0.0',
    config: firebaseModuleConfig,
  },
  {
    podName: 'EXFirebaseApp',
    libName: 'expo-firebase-app',
    sdkVersions: '>=31.0.0',
    config: firebaseModuleConfig,
  },
  {
    podName: 'EXFirebaseAuth',
    libName: 'expo-firebase-auth',
    sdkVersions: '>=31.0.0',
    config: firebaseModuleConfig,
  },
  {
    podName: 'EXFirebaseCrashlytics',
    libName: 'expo-firebase-crashlytics',
    sdkVersions: '>=31.0.0',
    config: firebaseModuleConfig,
  },
  {
    podName: 'EXFirebaseDatabase',
    libName: 'expo-firebase-database',
    sdkVersions: '>=31.0.0',
    config: firebaseModuleConfig,
  },
  {
    podName: 'EXFirebaseFirestore',
    libName: 'expo-firebase-firestore',
    sdkVersions: '>=31.0.0',
    config: firebaseModuleConfig,
  },
  {
    podName: 'EXFirebaseFunctions',
    libName: 'expo-firebase-functions',
    sdkVersions: '>=31.0.0',
    config: firebaseModuleConfig,
  },
  {
    podName: 'EXFirebaseInstanceID',
    libName: 'expo-firebase-instance-id',
    sdkVersions: '>=31.0.0',
    config: firebaseModuleConfig,
  },
  {
    podName: 'EXFirebaseInvites',
    libName: 'expo-firebase-invites',
    sdkVersions: '>=31.0.0',
    config: firebaseModuleConfig,
  },
  {
    podName: 'EXFirebaseLinks',
    libName: 'expo-firebase-links',
    sdkVersions: '>=31.0.0',
    config: firebaseModuleConfig,
  },
  {
    podName: 'EXFirebaseMessaging',
    libName: 'expo-firebase-messaging',
    sdkVersions: '>=31.0.0',
    config: firebaseModuleConfig,
  },
  {
    podName: 'EXFirebaseNotifications',
    libName: 'expo-firebase-notifications',
    sdkVersions: '>=31.0.0',
    config: firebaseModuleConfig,
  },
  {
    podName: 'EXFirebasePerformance',
    libName: 'expo-firebase-performance',
    sdkVersions: '>=31.0.0',
    config: firebaseModuleConfig,
  },
  {
    podName: 'EXFirebaseRemoteConfig',
    libName: 'expo-firebase-remote-config',
    sdkVersions: '>=31.0.0',
    config: firebaseModuleConfig,
  },
  {
    podName: 'EXFirebaseStorage',
    libName: 'expo-firebase-storage',
    sdkVersions: '>=31.0.0',
    config: firebaseModuleConfig,
  },
  {
    podName: 'EXFont',
    libName: 'expo-font',
    sdkVersions: '>=30.0.0',
  },
  {
    podName: 'EXFontInterface',
    libName: 'expo-font-interface',
    sdkVersions: '>=30.0.0',
  },
  {
    podName: 'EXGL',
    libName: 'expo-gl',
    sdkVersions: '>=29.0.0',
  },
  {
    podName: 'EXGL-CPP',
    libName: 'expo-gl-cpp',
    sdkVersions: '>=29.0.0',
    versionable: false,
    config: {
      ios: {
        subdirectory: 'cpp',
      },
    },
  },
  {
    podName: 'EXGoogleSignIn',
    libName: 'expo-google-sign-in',
    sdkVersions: '>=32.0.0',
  },
  {
    podName: 'EXHaptics',
    libName: 'expo-haptics',
    sdkVersions: '>=33.0.0',
  },
  {
    podName: 'EXImageLoaderInterface',
    libName: 'expo-image-loader-interface',
    sdkVersions: '>=30.0.0',
  },
  {
    podName: 'EXImageManipulator',
    libName: 'expo-image-manipulator',
    sdkVersions: '>=33.0.0',
  },
  {
    podName: 'EXImagePicker',
    libName: 'expo-image-picker',
    sdkVersions: '>=33.0.0',
  },
  {
    libName: 'expo-intent-launcher',
    sdkVersions: '>=33.0.0',
    config: {
      ios: {
        versionable: false,
        detachable: false,
        includeInExpoClient: false,
      },
    },
  },
  {
    podName: 'EXKeepAwake',
    libName: 'expo-keep-awake',
    sdkVersions: '>=33.0.0',
  },
  {
    podName: 'EXLinearGradient',
    libName: 'expo-linear-gradient',
    sdkVersions: '>=33.0.0',
  },
  {
    podName: 'EXLocalAuthentication',
    libName: 'expo-local-authentication',
    sdkVersions: '>=30.0.0',
  },
  {
    podName: 'EXLocalization',
    libName: 'expo-localization',
    sdkVersions: '>=31.0.0',
  },
  {
    podName: 'EXLocation',
    libName: 'expo-location',
    sdkVersions: '>=30.0.0',
  },
  {
    podName: 'EXMailComposer',
    libName: 'expo-mail-composer',
    sdkVersions: '>=33.0.0',
  },
  {
    podName: 'EXMediaLibrary',
    libName: 'expo-media-library',
    sdkVersions: '>=30.0.0',
  },
  {
    podName: 'EXPaymentsStripe',
    libName: 'expo-payments-stripe',
    sdkVersions: '>=30.0.0',
    detachable: false,
    config: {
      ios: {
        includeInExpoClient: false,
      },
    },
  },
  {
    podName: 'EXPermissions',
    libName: 'expo-permissions',
    sdkVersions: '>=29.0.0',
  },
  {
    podName: 'EXPermissionsInterface',
    libName: 'expo-permissions-interface',
    sdkVersions: '>=29.0.0',
  },
  {
    podName: 'EXPrint',
    libName: 'expo-print',
    sdkVersions: '>=30.0.0',
  },
  {
    podName: 'EXReactNativeAdapter',
    libName: 'expo-react-native-adapter',
    sdkVersions: '>=29.0.0',
  },
  {
    podName: 'EXSecureStore',
    libName: 'expo-secure-store',
    sdkVersions: '>=33.0.0',
  },
  {
    podName: 'EXSensors',
    libName: 'expo-sensors',
    sdkVersions: '>=29.0.0',
  },
  {
    podName: 'EXSensorsInterface',
    libName: 'expo-sensors-interface',
    sdkVersions: '>=29.0.0',
  },
  {
    podName: 'EXSMS',
    libName: 'expo-sms',
    sdkVersions: '>=29.0.0',
  },
  {
    podName: 'EXSpeech',
    libName: 'expo-speech',
    sdkVersions: '>=33.0.0',
  },
  {
    podName: 'EXSQLite',
    libName: 'expo-sqlite',
    sdkVersions: '>=33.0.0',
  },
  {
    podName: 'EXTaskManager',
    libName: 'expo-task-manager',
    sdkVersions: '>=32.0.0',
  },
  {
    podName: 'EXTaskManagerInterface',
    libName: 'expo-task-manager-interface',
    sdkVersions: '>=32.0.0',
    config: {
      android: {
        versionable: false,
      },
    },
  },
  {
    podName: 'EXWebBrowser',
    libName: 'expo-web-browser',
    sdkVersions: '>=33.0.0',
  },

  // JS-only modules
  {
    libName: 'expo-asset',
    sdkVersions: '>=29.0.0',
    isNativeModule: false,
  },
  {
    libName: 'expo-module-template',
    sdkVersions: '>=29.0.0',
    isNativeModule: false,
  },
];

function defaults(defaultConfig, ...customConfigs) {
  const config = { ...defaultConfig };
  for (const customConfig of customConfigs) {
    if (customConfig) {
      Object.assign(config, customConfig || {});
    }
  }
  return config;
}

const expoSdkUniversalModulesConfigs = expoSdkUniversalModules.map(
  ({ config, podName, libName, sdkVersions, isNativeModule, ...params }) => {
    return {
      podName,
      libName,
      sdkVersions,
      isNativeModule: isNativeModule == null ? true : isNativeModule,
      config: {
        ios: defaults(defaultUniversalModuleConfig.ios, params, config && config.ios),
        android: defaults(defaultUniversalModuleConfig.android, params, config && config.android),
      },
    };
  }
);

module.exports = {
  expoSdkUniversalModulesConfigs,
};
