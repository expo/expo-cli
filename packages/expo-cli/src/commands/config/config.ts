import { getConfig, ProjectConfig } from '@expo/config';
import { compileModsAsync } from '@expo/config-plugins/build/plugins/mod-compiler';
import { getPrebuildConfig } from '@expo/prebuild-config';
import type { Command } from 'commander';

import CommandError from '../../CommandError';
import Log from '../../log';
import { logConfig } from '../eject/configureProjectAsync';
import { profileMethod } from '../utils/profileMethod';

type Options = {
  type?: string;
  full: boolean;
};

async function actionAsync(projectRoot: string, options: Options) {
  let config: ProjectConfig;

  if (options.type === 'prebuild') {
    config = profileMethod(getPrebuildConfig)(projectRoot, {
      platforms: ['ios', 'android'],
    });
  } else if (options.type === 'introspect') {
    config = profileMethod(getPrebuildConfig)(projectRoot, {
      platforms: ['ios', 'android'],
    });

    await compileModsAsync(config.exp, {
      projectRoot,
      introspect: true,
      platforms: ['ios', 'android'],
      assertMissingModProviders: false,
    });
    // @ts-ignore
    delete config.modRequest;
    // @ts-ignore
    delete config.modResults;
  } else if (options.type === 'public') {
    config = profileMethod(getConfig)(projectRoot, {
      skipSDKVersionRequirement: true,
      isPublicConfig: true,
    });
  } else if (options.type) {
    throw new CommandError(
      `Invalid option: --type ${options.type}. Valid options are: public, prebuild`
    );
  } else {
    config = profileMethod(getConfig)(projectRoot, {
      skipSDKVersionRequirement: true,
    });
  }

  Log.log();
  logConfig(options.full ? config : config.exp);
  Log.log();
}

export default function (program: Command) {
  program
    .command('config [path]')
    .description('Show the project config')
    .helpGroup('info')
    .option('-t, --type <type>', 'Type of config to show. Options: public, prebuild, introspect')
    .option('--full', 'Include all project config data')
    .asyncActionProjectDir(profileMethod(actionAsync));
}
