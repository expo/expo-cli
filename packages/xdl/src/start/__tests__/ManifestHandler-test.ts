import { ExpoConfig } from '@expo/config';
import axios from 'axios';
import fs from 'fs-extra';
import { vol } from 'memfs';
import os from 'os';
import path from 'path';
import uuid from 'uuid';

import {
  getManifestResponseAsync,
  getSignedManifestStringAsync,
  getUnsignedManifestString,
} from '../ManifestHandler';

const actualFs = jest.requireActual('fs') as typeof fs;
jest.mock('fs');
jest.mock('axios');

jest.mock('../../User', () => {
  // const user = jest.requireActual('../../User');
  return {
    // ...user,
    ANONYMOUS_USERNAME: 'anonymous',
    getSessionAsync() {
      return null;
    },
    ensureLoggedInAsync: () => ({
      sessionSecret: 'SECRET',
    }),
  };
});
jest.mock('../../project/ExpSchema', () => {
  // const user = jest.requireActual('../../User');
  return {
    getAssetSchemasAsync() {
      return ['icon', 'splash.image'];
    },
  };
});

const mockManifest: ExpoConfig = {
  name: 'Hello',
  slug: 'hello-world',
  owner: 'ownername',
  version: '1.0.0',
  platforms: ['ios'],
};

const mockSignedManifestResponse = JSON.stringify({
  manifestString: JSON.stringify(mockManifest),
  signature: 'SIGNATURE_HERE',
  version: '1.0.0',
});

describe('getSignedManifestStringAsync', () => {
  it('calls the server API to sign a manifest', async () => {
    const requestFunction = axios.request as jest.MockedFunction<typeof axios.request>;
    requestFunction.mockReturnValueOnce(
      Promise.resolve({
        status: 200,
        data: { data: { response: mockSignedManifestResponse } },
      })
    );
    const manifestString = await getSignedManifestStringAsync(mockManifest, {
      sessionSecret: 'SECRET',
    });
    expect(manifestString).toBe(mockSignedManifestResponse);
    expect(requestFunction.mock.calls[0][0]).toMatchInlineSnapshot(`
      Object {
        "data": Object {
          "args": Object {
            "remotePackageName": "hello-world",
            "remoteUsername": "ownername",
          },
          "manifest": Object {
            "name": "Hello",
            "owner": "ownername",
            "platforms": Array [
              "ios",
            ],
            "slug": "hello-world",
            "version": "1.0.0",
          },
        },
        "headers": Object {
          "Expo-Session": "SECRET",
          "Exponent-Client": "xdl",
        },
        "maxBodyLength": 104857600,
        "maxContentLength": 104857600,
        "method": "post",
        "url": "https://exp.host/--/api/v2/manifest/sign",
      }
    `);
  });
});

describe('getUnsignedManifestString', () => {
  it('returns a stringified manifest with the same shape a server-signed manifest', () => {
    expect(getUnsignedManifestString(mockManifest)).toMatchSnapshot();
  });
});

describe('getManifestResponseAsync', () => {
  // for some reason, tempy fails with memfs in XDL
  const expoDir = path.join(os.tmpdir(), `.expo-${uuid.v4()}`);

  beforeAll(() => {
    process.env.__UNSAFE_EXPO_HOME_DIRECTORY = expoDir;
    fs.mkdirpSync(expoDir);
  });

  afterAll(() => {
    process.env.__UNSAFE_EXPO_HOME_DIRECTORY = '';
    fs.removeSync(expoDir);
  });

  beforeEach(() => {
    const packageJson = JSON.stringify(
      {
        name: 'testing123',
        version: '0.1.0',
        main: 'index.js',
      },
      null,
      2
    );

    const appJson = JSON.stringify(
      {
        expo: {
          name: 'testing 123',
          icon: './icon.png',
          splash: { image: './assets/splash.png' },
          version: '0.1.0',
          sdkVersion: '38.0.0',
          slug: 'testing-123',
          extras: { myExtra: '123' },
        },
      },
      null,
      2
    );

    vol.fromJSON({
      '/alpha/package.json': packageJson,
      '/alpha/app.json': appJson,
      '/alpha/index.js': 'console.log("lol")',
    });

    const iconPath = path.resolve(__dirname, './fixtures/icon.png');
    const icon = actualFs.readFileSync(iconPath);
    vol.mkdirpSync('/alpha/assets');
    vol.writeFileSync('/alpha/icon.png', icon);
    vol.writeFileSync('/alpha/assets/splash.png', icon);
  });

  afterEach(() => {
    vol.reset();
    delete process.env.EXPO_SOME_TEST_VALUE;
    delete process.env.REACT_NATIVE_TEST_VALUE;
    delete process.env.EXPO_APPLE_PASSWORD;
  });

  it(`returns a minimal expected manifest`, async () => {
    process.env.EXPO_SOME_TEST_VALUE = 'true';
    process.env.REACT_NATIVE_TEST_VALUE = 'true';
    process.env.EXPO_APPLE_PASSWORD = 'my-password';

    const res = await getManifestResponseAsync({
      projectRoot: '/alpha',
      host: '127.0.0.1:19000',
      platform: 'ios',
      acceptSignature: 'true',
    });

    // console.log(res.exp);

    // Values starting with EXPO_ or REACT_NATIVE_ get added to the env and exposed to expo-constants
    expect(res.exp.env.EXPO_SOME_TEST_VALUE).toBe('true');
    expect(res.exp.env.REACT_NATIVE_TEST_VALUE).toBe('true');
    // This value is blacklisted
    expect(res.exp.env.EXPO_APPLE_PASSWORD).not.toBeDefined();
    // Users should use app.config.js + extras now so test that it always works
    expect(res.exp.extras.myExtra).toBe('123');

    // Ensure the bundle URL is built correctly
    expect(res.exp.bundleUrl).toBe(
      'http://127.0.0.1:80/index.bundle?platform=ios&dev=true&hot=false&minify=false'
    );
    expect(res.exp.debuggerHost).toBe('127.0.0.1:80');
    expect(res.exp.logUrl).toBe('http://127.0.0.1:80/logs');
    expect(res.exp.hostUri).toBe('127.0.0.1:80');

    expect(res.exp.mainModuleName).toBe('index');
    expect(res.exp.packagerOpts).toBeDefined();
    // Required for various tools
    expect(res.exp.developer.projectRoot).toBe('/alpha');

    // ProjectAssets gathered URLs
    expect(res.exp.iconUrl).toBe('http://127.0.0.1:80/assets/./icon.png');
    expect(res.exp.splash.imageUrl).toBe('http://127.0.0.1:80/assets/./assets/splash.png');
  });
});
