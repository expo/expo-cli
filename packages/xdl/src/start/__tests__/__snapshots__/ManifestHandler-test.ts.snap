// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getUnsignedManifestString returns a stringified manifest with the same shape a server-signed manifest 1`] = `"{\\"manifestString\\":\\"{\\\\\\"name\\\\\\":\\\\\\"Hello\\\\\\",\\\\\\"slug\\\\\\":\\\\\\"hello-world\\\\\\",\\\\\\"owner\\\\\\":\\\\\\"ownername\\\\\\",\\\\\\"version\\\\\\":\\\\\\"1.0.0\\\\\\",\\\\\\"platforms\\\\\\":[\\\\\\"ios\\\\\\"]}\\",\\"signature\\":\\"UNSIGNED\\"}"`;
